core/templates/pages/contributor-dashboard-page/translation-language-selector/review-translation-language-selector.component.spec.ts

```diff
@@ -97,209 +107,241 @@ describe('Review Translation language selector', () => {
     component.activeLanguageCode = 'en';
   });
 
-  it('should correctly initialize languageIdToDescription map', () => {
-    expect(component.languageIdToDescription.en).toBe('English');
-    expect(component.languageIdToDescription.fr).toBe('français (French)');
-  });
-
-  it('should correctly initialize dropdown activeLanguageCode', () => {
-    const dropdown = (
-      fixture.nativeElement.querySelector(
-        '.oppia-review-translation-language-selector-inner-container'));
+  describe('when reviewer translation rights are found', () => {
+    const translationReviewerLanguageCodes = ['en', 'es', 'fr'];
 
-    expect(dropdown.firstChild.textContent.trim()).toBe('English');
-  });
+    beforeEach(fakeAsync(() => {
+      spyOn(userService, 'getUserContributionRightsDataAsync').and.resolveTo({
+        can_suggest_questions: false,
+        can_review_translation_for_language_codes: (
+          translationReviewerLanguageCodes),
+        can_review_voiceover_for_language_codes: [],
+        can_review_questions: false
+      });
+      component.ngOnInit();
+      tick();
+      fixture.detectChanges();
+    }));
 
-  it('should correctly show and hide the dropdown', fakeAsync(() => {
-    expect(component.dropdownShown).toBe(false);
-    expect(getDropdownOptionsContainer()).toBeFalsy();
+    it('should correctly initialize languageIdToDescription map', () => {
```

Comment:
To be more specific: 
```suggestion
    it('should initialize languageIdToDescription map to user's reviewable languages', () => {
```